### ISSUE TITLE
Feature request : in markdown use todotxt codeblocks instead of "Tasks" header

### ISSUE THREAD

**Issue by jpbourgeon** [davraamides/todotxt-mode#16]
Hi

I use your extension extensively. thanks for the good work.

In my opinion, in markdown files, it would be more aligned with both languages specifications to merge todotxt tasks into codeblocks instead of a tasks section. For example:

Replace this :

```
## Tasks
(A) @context +project task due:date
(B) @context +project2 task2 due:date
x YYYY-MM-DD @context2 +project2 task3 key:value
```

by this (I replaced the codeblocks backticks by singlequotes but you get it)

```
'''todotxt
(A) @context +project task due:date
(B) @context +project2 task2 due:date
x YYYY-MM-DD @context2 +project2 task3 key:value
'''
```

I see 3 reasons for this change:

- todotxt is not in the markdown specification and codeblocks is the official way to insert another code language into markdown
- there are markdown conventions incompatible with your implementation. I use a markdown linter that throws an error when there is not a blank line between a header and its first child content...
- Freedom and i18n ! Your tasks codeblocks would not be dependent of a fixed naming anymore. I could name my tasks section "Tâches à faire", "Todo list" or whatever.

It may be an opportunity to isolate the tasks inside markdown more efficiently to make all the good stuff of your extension available to todotxt codeblocks in markdown too (like sorting, tagging priorities, etc.) and even provide new opportunities like having multiple tasks sections in the same document with power comments before the codeblock `<!-- todotxt:task_filtering_regex -->` For an example:

```
## Project Todolist

<!-- todotxt:^.*\+project .*$ -->
'''todotxt
(A) @context +project task due:date
'''

## Project 2 completed tasks

<!-- todotxt:^x.*\+project2 .*$ -->
'''todotxt
x YYYY-MM-DD @context2 +project2 task3 key:value
'''
```

Thanks for considering this !

**Comment by davraamides** [comment]
Thanks for the interesting suggestions, Jean-Philippe. I've actually been jammed with work since I started living in lockdown a month ago so haven't had time to look at this but hopefully things will get more normal and I'll have time to look at this more carefully in the near future.

Note that the Markdown integration simply uses two regular expressions: `todotxtmode.markdownDecorationBeginPattern` and `todotxtmode.markdownDecorationEndPattern`, which have defaults of using a "Tasks" heading. Changing them to use a special codeblock pattern should work fine (as well as changing them to use "Tâches à faire" as you mention).

In terms of support of the other features within that region (sorting, priorities, etc.) I'll have to look and see how good a job I did generalizing the code!

**Comment by jpbourgeon** [comment]
Wonderful @davraamides !

I should have read the docs more carefully (RTFM) :-)

In my `.code-workspace` :

```json
{
  //...
  "todotxtmode.markdownDecorationBeginPattern": "^\\`\\`\\`todotxt$",
  "todotxtmode.markdownDecorationEndPattern": "^\\`\\`\\`$",
}
```

Works like a charm 👍

I tried a few shortcut inside a code block (priority, completing task, archiving...) and it works quite well

**Comment by davraamides** [comment]
That's great. I'm going to close this for now. If you find problems, feel free to open another issue.

**Comment by emesar** [comment]
Hi - sorry for resurrecting an old thread, but for whatever reason I can't get the `todotxtmode.markdownDecorationBeginPattern` and `todotxtmode.markdownDecorationEndPattern` settings to work. I've tried a variety of different RegEx patterns (including the default and the code block example above) without success. Is this feature no longer supported? Or am I doing something horribly wrong? Appreciate any help!

**Comment by davraamides** [comment]
Hi emesar. Yes, this still works. I've pasted an image below showing an example. In addition to the two settings you mentioned, there is also the `todotxtmode.markdownFilePattern` pattern that defines the Markdown filenames that should have this behavior enabled. The default is `^.*\\.(md|markdown)$` which would match files ending in `.md` or in `.markdown`. Is it possible you are naming yours differently?

<img width="774" alt="Screen Shot 2021-02-26 at 11 20 08 AM" src="https://user-images.githubusercontent.com/1895577/109325996-9d691f00-7824-11eb-8cad-c8338d549d7e.png">

**Comment by emesar** [comment]
Hi @davraamides, thanks for the quick response. Based on your feedback, I realized my issue might be with the specific RegEx flavor being used. Ended up discovering that while this _didn't_ work:
```json
{
  "todotxtmode.markdownDecorationBeginPattern": "^\\`\\`\\`todotxt$",
  "todotxtmode.markdownDecorationEndPattern": "^\\`\\`\\`$",
}
```
This _did_ work:
```json
{
  "todotxtmode.markdownDecorationBeginPattern": "^```todotxt$",
  "todotxtmode.markdownDecorationEndPattern": "^```$",
}
```
With that knowledge, I settled on the patterns I actually wanted:
![image](https://user-images.githubusercontent.com/56656458/109331964-7bbf6600-782b-11eb-928c-e2e8742dd2e4.png)

Seems to work perfectly now. Thanks for your help, and all of your great work on this extension!
